source-similarity-matrix.py

```diff
@@ -9,49 +9,62 @@
 import tensorflow_text
 from structlog import get_logger
 from tqdm import tqdm
+import multiprocessing
 
 import config
 from utils import download_file, upload_file
 
 logger = get_logger()
 
 
-def embed(input):
-    return model(input)
-
+def upload_source_sim_files(lang_region):
+    logger.info("Uploading sources sim files")
+    if not config.NO_UPLOAD:
+        upload_file(config.OUTPUT_DIR + "/" + f'/{config.SOURCE_SIMILARITY_T10.format(LANG_REGION=lang_region)}.json',
+                    config.PUB_S3_BUCKET,
+                    f"source-suggestions/{config.SOURCE_SIMILARITY_T10.format(LANG_REGION=lang_region)}.json")
 
-# Take centroid of 512-d embeddings
-def get_source_representation_from_titles(titles):
-    source_repr = np.zeros((1, 512))
-    if len(titles) < config.MINIMUM_ARTICLE_HISTORY_SIZE:
-        return source_repr
+        upload_file(
+            config.OUTPUT_DIR + "/" + f'/{config.SOURCE_SIMILARITY_T10_HR.format(LANG_REGION=lang_region)}.json',
+            config.PUB_S3_BUCKET,
+            f"source-suggestions/{config.SOURCE_SIMILARITY_T10_HR.format(LANG_REGION=lang_region)}.json")
 
-    for title in titles:
-        source_repr += embed([title])[0]
-    norm_repr = tf.nn.l2_normalize(source_repr / len(titles), axis=1)
-    return norm_repr.numpy()
+        upload_file(config.OUTPUT_DIR + "/" + f'/{config.SOURCE_EMBEDDINGS.format(LANG_REGION=lang_region)}.csv',
+                    config.PUB_S3_BUCKET,
+                    f"source-suggestions/{config.SOURCE_EMBEDDINGS.format(LANG_REGION=lang_region)}.csv")
 
 
-def compute_source_similarity(source1, source2, t='dot'):
-    cosine_similarities = np.dot(source1, np.transpose(source2))
-    clip_cosine_similarity = tf.clip_by_value(cosine_similarities, -1.0, 1.0)
-    score = 1.0 - tf.acos(clip_cosine_similarity) / math.pi
-    return score
+def compute_sources_representations(model_lang):
```

Comment:
would call this function something like `update_source_sim_files`

Reply:
Done